concordium-consensus/src/Concordium/GlobalState/Persistent/Genesis.hs

```diff
@@ -0,0 +1,274 @@
+
+        return $
+            BS.PersistentBirkParameters
+                { _birkCurrentEpochBakers = _birkNextEpochBakers,
+                  _birkSeedState = Types.initialSeedState genesisLeadershipElectionNonce genesisEpochLength,
+                  ..
+                }
+
+    -- Reward details
+    rewardDetails <- case Types.delegationSupport @av of
+        Types.SAVDelegationNotSupported ->
+            return $ BS.BlockRewardDetailsV0 BS.emptyHashedEpochBlocks
+        Types.SAVDelegationSupported ->
+            case Types.delegationChainParameters @pv of
+                Types.DelegationChainParametersV1 -> do
+                    capRef :: Blob.HashedBufferedRef CapDist.CapitalDistribution <-
+                        Blob.refMakeFlushed
+                            CapDist.CapitalDistribution
+                                { bakerPoolCapital = agsBakerCapitals,
+                                  passiveDelegatorsCapital = Vec.empty
+                                }
+                    bakerPoolRewardDetails <-
+                        LFMBT.fromAscList $
+                            replicate (Vec.length agsBakerCapitals) Basic.emptyBakerPoolRewardDetails
+                    BS.BlockRewardDetailsV1
+                        <$> Blob.refMakeFlushed
+                            Rewards.PoolRewards
+                                { nextCapital = capRef,
+                                  currentCapital = capRef,
+                                  bakerPoolRewardDetails = bakerPoolRewardDetails,
+                                  passiveDelegationTransactionRewards = 0,
+                                  foundationTransactionRewards = 0,
+                                  nextPaydayEpoch =
+                                    genesisChainParameters
+                                        ^. Types.cpTimeParameters
+                                            . Types.tpRewardPeriodLength
+                                            . to Types.rewardPeriodEpochs,
+                                  nextPaydayMintRate =
+                                    genesisChainParameters
+                                        ^. Types.cpTimeParameters
+                                            . Types.tpMintPerPayday
+                                }
+
+    -- Module
+    modules <- Blob.refMakeFlushed Modules.emptyModules
+
+    -- Identity providers and anonymity revokers
+    identityProviders <- Blob.bufferHashed $ Types.makeHashed genesisIdentityProviders
+    anonymityRevokers <- Blob.bufferHashed $ Types.makeHashed genesisAnonymityRevokers
+
+    cryptographicParameters <- Blob.bufferHashed $ Types.makeHashed genesisCryptographicParameters
+
+    persistentUpdates <- Updates.initialUpdates genesisUpdateKeys genesisChainParameters
+    updates <- Blob.makeBufferedRef persistentUpdates
+
+    releaseSchedule <- ReleaseSchedule.emptyReleaseSchedule
+    bsp <-
+        Blob.makeBufferedRef $
```

Comment:
Is this deliberately not flushed?

It does not matter since it is flushed quickly after this function returns, but it would be good to document intent clearly.

Reply:
I'll make this flush as well